ocaml/xapi/storage_mux.ml

```diff
@@ -718,19 +722,8 @@ module Mux = struct
         (Hashtbl.find plugins sr).backend_domain
   end
 
-  module TASK = struct
-    let stat () ~dbg:_ ~task:_ = assert false
-
-    let cancel () ~dbg:_ ~task:_ = assert false
-
-    let destroy () ~dbg:_ ~task:_ = assert false
-
-    let list () ~dbg:_ = assert false
-  end
-
-  module UPDATES = struct
-    let get () ~dbg:_ ~from:_ ~timeout:_ = assert false
-  end
+  module TASK = Storage_smapiv1_wrapper.Impl.TASK
```

Comment:
there is a smapiv3 task api, but I don't think any SRs actually implement it currently (it is meant to run things like the SMGC in a way that is not totally opaque to XAPI, or long running tasks such as mirroring/etc.).
I don't know how useful it'd be, maybe @MarkSymsCtx knows whether we intend to use the SMAPIv3 tasks API.

Reply:
I never even knew there was such a thing.

Reply:
https://xapi-project.github.io/xapi-storage/#task-errors I don't know whether it works at all, probably hooked up in all the right places in XAPI, but if it is useful you can using it.